invoices/invoiceregistry.go

```diff
@@ -1613,10 +1612,10 @@ func (i *InvoiceRegistry) SubscribeSingleInvoice(
 	}
 	client.ntfnQueue.Start()
 
-	i.clientMtx.Lock()
+	// Always increment by 1 first, and our client ID will start with 1,
+	// not 0.
+	atomic.AddUint32(&i.nextClientID, 1)
```

Comment:
Use return value here as well to make truly atomic.

Reply:
Fixed.